hugolib/page.go

```diff
@@ -319,12 +308,20 @@ func (p *pageState) Resources() resource.Resources {
 	return p.resources
 }
 
-func (p *pageState) HasShortcode(name string) bool {
+// HasShortcode returns whether the page has a given named shortcode or any
+// shortcodes at all.  To see if any shortcodes are present, do not pass a name
+// parameter.
```

Comment:
This behaviour does not feel natural to me. I would say that:

* HasShortcode nil => true if any shortcodes at all
* HasShortcode a b => true if both a and be exists

Also:

* All the whitespace/formatting changes I assume does not come from `gofmt`?
* I don't see any tests for this new logic.